Source/SwiftLintFramework/Rules/Lint/DiscardedNotificationCenterObserverRule.swift

```diff
@@ -78,6 +82,11 @@ public struct DiscardedNotificationCenterObserverRule: ASTRule, ConfigurationPro
             return []
         }
 
+        let kinds = file.structureDictionary.kinds(forByteOffset: offset)
+        if kinds.count >= 2, kinds[kinds.count - 2].0 == "source.lang.swift.expr.array" {
```

Comment:
```suggestion
        if kinds.count >= 2 && SwiftExpressionKind(rawValue: kinds[kinds.count - 2].0) == .array {
```

Reply:
Thanks for the suggestions. I have updated the commit with fixes to these issues.